samza-core/src/main/java/org/apache/samza/config/JobCoordinatorConfig.java

```diff
@@ -37,4 +38,14 @@ public String getJobCoordinatorFactoryClassName() {
 
     return jobCoordinatorFactoryClassName;
   }
+
+  public String getJobCoordinationServiceFactoryClassName() {
+    String jobCooridanationFactoryClassName = get(JOB_COORDINATIOIN_SERVICE_FACTORY, "org.apache.samza.zk.ZkCoordinationServiceFactory");
+    if (Strings.isNullOrEmpty(jobCooridanationFactoryClassName)) {
+      throw new ConfigException(
+          String.format("Missing config - %s. Cannot start coordiantion service!", JOB_COORDINATOR_FACTORY));
```

Comment:
Is this exception message accurate? If we have the default Factory class? Then it should not be caused by the missing config because missing config is acceptable.